src/routes/class-base-buckaroo-route.php

```diff
@@ -0,0 +1,154 @@
+
+  /**
+   * Construct object
+   *
+   * @param Config_Data                     $config                          Config data obj.
+   * @param Buckaroo                        $buckaroo                        Buckaroo integration obj.
+   * @param Buckaroo_Response_Handler_Route $buckaroo_response_handler_route Response handler route obj.
+   * @param Authorization_Interface         $hmac                            Authorization object.
+   * @param Basic_Captcha                   $basic_captcha                   Basic_Captcha object.
+   */
+  public function __construct(
+    Config_Data $config,
+    Buckaroo $buckaroo,
+    Buckaroo_Response_Handler_Route $buckaroo_response_handler_route,
+    Authorization_Interface $hmac,
+    Basic_Captcha $basic_captcha
+  ) {
+    $this->config                          = $config;
+    $this->buckaroo                        = $buckaroo;
+    $this->buckaroo_response_handler_route = $buckaroo_response_handler_route;
+    $this->hmac                            = $hmac;
+    $this->basic_captcha                   = $basic_captcha;
+  }
+
+  /**
+   * We need to define redirect URLs so that Buckaroo redirects the user to our buckaroo-response-handler route
+   * which might run some custom logic and then redirect the user to the actual redirect URL as defined in the form's
+   * options.
+   *
+   * @param array $params Array of WP_REST_Request params.
+   * @return array
+   */
+  protected function set_redirect_urls( array $params ): array {
+
+    // Now let's define all Buckaroo-recognized statuses for which we need to provide redirect URLs.
+    $statuses = [
+      Buckaroo_Response_Handler_Route::STATUS_SUCCESS,
+      Buckaroo_Response_Handler_Route::STATUS_CANCELED,
+      Buckaroo_Response_Handler_Route::STATUS_ERROR,
+      Buckaroo_Response_Handler_Route::STATUS_REJECT,
+    ];
+
+    // Now let's build redirect URLs (to buckaroo-response-handler middleware route) for each status.
+    $redirect_urls = [];
+    $base_url      = \home_url( $this->buckaroo_response_handler_route->get_route_uri() );
+    foreach ( $statuses as $status_value ) {
+      $url_params = $params;
+      $url_params[ Buckaroo_Response_Handler_Route::STATUS_PARAM ] = $status_value;
+      $url = \add_query_arg( array_merge(
+        $url_params,
+        [ HMAC::AUTHORIZATION_KEY => rawurlencode( $this->hmac->generate_hash( $url_params, $this->generate_authorization_salt_for_response_handler() ) ) ]
+      ), $base_url );
+
+      $redirect_urls[] = $url;
+    }
+
+    $this->buckaroo->set_redirect_urls( ...$redirect_urls );
+
+    return $params;
+  }
+
+  /**
+   * Set Buckaroo to test mode if test param provided.
+   *
+   * @param array $params Request params.
+   * @return void
+   */
+  protected function set_test_if_needed( array $params ): void {
+    if ( isset( $params[ self::TEST_PARAM ] ) && filter_var( $params[ self::TEST_PARAM ], FILTER_VALIDATE_BOOL ) ) {
```

Comment:
What's the usage of `filter_var( $params[ self::TEST_PARAM ], FILTER_VALIDATE_BOOL )` for?

Reply:
This is used to allow triggering a test Buckaroo response from Form editor (by adding a hidden field `test` with value `1`). Rather than requiring that the value is `=== 1`, by using this filter the editor can set it however they like and it will work. It will work if they put the value as `1`, ` 1`, `yes`, `true`, etc.

Reply:
Cool!